src/rustup-cli/help.rs

```diff
@@ -78,7 +78,23 @@ inferred, so the above could be written:
     $ rustup default stable-msvc
 
 Toolchain names that don't name a channel instead can be used to name
-custom toolchains with the `rustup toolchain link` command.";
+custom toolchains with the `rustup toolchain link` command. This can
+also be used to symlink toolchains from local builds. For more
+information see `rustup toolchain help link`.";
+
+pub static TOOLCHAIN_LINK_HELP: &'static str =
+r"
+Symlinks a toolchain from a local directory specified by 'path' and
+gives it a custom name specified by 'toolchain' if it does not name a
+standard release channel.
```

Comment:
> if that's not already the case

Currently, `TOOLCHAIN_HELP` (used for `rustup help toolchain`) only tells what each subcommand does in a succinct manner and each subcommand is responsible for its own documentation (or lack thereof). The current documentation for `link` inside `TOOLCHAIN_HELP` is just:

> SUBCOMMANDS:
> ...
>     link      Create a custom toolchain by symlinking to a directory
> ...
> Toolchain names that don't name a channel instead can be used to name
> custom toolchains with the `rustup toolchain link` command. This can
> also be used to symlink toolchains from local builds. For more
> information see `rustup toolchain help link`.

I feel that this is sufficient information to indicate to the user whether `toolchain link` is something they want to use, although the wording could be improved to make the dual purpose (overwriting or aliasing) clearer. Just making the utility clearer doesn't require the full details; e.g. how the path is to be specified.